control-plane/pkg/reconciler/channel/v2/channelv2.go

```diff
@@ -576,40 +596,61 @@ func (r Reconciler) reconcileConsumerGroup(ctx context.Context, channel *messagi
 		},
 	}
 
+	expectedCg.Spec.Replicas = ptr.Int32(1) //Must be set for Scaled Object ref
```

Comment:
I think this will end up overriding whatever KEDA has set because 
```
if equality.Semantic.DeepDerivative(expectedCg.Spec, cg.Spec) ...
```
will be always false.

If it's not done yet, I think this should be done at the webhook level if replicas is not set.

Reply:
Don't think its done at the webhook level today. Good pt